vmdb/app/controllers/application_controller/explorer.rb

```diff
@@ -927,8 +927,7 @@ def x_get_tree_custom_kids(object, options)
       return count_only ? objects.length : objects
     when :alert_profile
       # Add all alert profiles so links back from Alerts etc will work - TODO: figure out how to load on demand
-      objects = MiqAlertSet.all(:conditions=> ["(mode = ?)", object[:id].split('-')],
-                          :order => "lower(description) ASC")
+      objects = MiqAlertSet.where(:mode => object[:id].split('-')).order("lower(description) ASC")
```

Comment:
One look at it says it's definitely on someone's rewrite list :smile:

But, there are clearly already tests hitting this line (otherwise I wouldn't've seen the warning). And having encountered a couple of those `delete` cases.... I've not found them to be particularly subtle.

It's true we've done a reasonable amount of just-make-it-`to_a` already, but I'm not convinced that's a good reason to continue; new code should be working with relations, and the only way to get to there is to start doing it.

Ultimately, my concern is that if we don't go for it now, *while* everyone's on the lookout for "blame it on rails4", we'll end up permanently stuck having to eagerly make everything an array, just in case it might find its way back to older code that might possibly not be expecting it.